Source/Scene/Cesium3DTileGroundPrimitive.js

```diff
@@ -0,0 +1,875 @@
+        this._sp = undefined;
+        this._spPick = undefined;
+
+        this._rsStencilPreloadPass = undefined;
+        this._rsStencilDepthPass = undefined;
+        this._rsColorPass = undefined;
+        this._rsPickPass = undefined;
+
+        this._commands = [];
+        this._pickCommands = [];
+    }
+
+    var attributeLocations = {
+        position : 0,
+        a_batchId : 1
+    };
+
+    var scratchDecodeMatrix = new Matrix4();
+    var scratchEncodedPosition = new Cartesian3();
+    var scratchNormal = new Cartesian3();
+    var scratchScaledNormal = new Cartesian3();
+    var scratchMinHeightPosition = new Cartesian3();
+    var scratchMaxHeightPosition = new Cartesian3();
+    var scratchBVCartographic = new Cartographic();
+    var scratchBVRectangle = new Rectangle();
+
+    function createVertexArray(primitive, context) {
+        if (!defined(primitive._positions)) {
+            return;
+        }
+
+        var positions = primitive._positions;
+        var offsets = primitive._offsets;
+        var counts = primitive._counts;
+        var indexOffsets = primitive._indexOffsets;
+        var indexCounts = primitive._indexCounts;
+        var indices = primitive._indices;
+        var boundingVolumes = primitive._boundingVolumes;
+        var center = primitive._center;
+        var ellipsoid = primitive._ellispoid;
+
+        var minHeight = primitive._minimumHeight;
+        var maxHeight = primitive._maximumHeight;
+
+        var quantizedOffset = primitive._quantizedOffset;
+        var quantizedScale = primitive._quantizedScale;
+        var decodeMatrix = Matrix4.fromTranslationRotationScale(new TranslationRotationScale(quantizedOffset, undefined, quantizedScale), scratchDecodeMatrix);
+
+        var positionsLength = positions.length;
+        var batchedPositions = new Float32Array(positionsLength * 2.0);
+        var batchedIds = new Uint16Array(positionsLength / 3 * 2);
+        var batchedIndexOffsets = new Array(indexOffsets.length);
+        var batchedIndexCounts = new Array(indexCounts.length);
+
+        // TODO: compute length and create typed array
+        var batchedIndices = [];
+
+        var colors = primitive._colors;
+        var colorsLength = colors.length;
+
+        var i;
+        var j;
+        var color;
+        var rgba;
+
+        var buffers = {};
+        for (i = 0; i < colorsLength; ++i) {
+            rgba = colors[i].toRgba();
+            if (!defined(buffers[rgba])) {
+                buffers[rgba] = {
+                    positionLength : counts[i],
+                    indexLength : indexCounts[i],
+                    offset : 0,
+                    indexOffset : 0,
+                    batchIds : [i]
+                };
+            } else {
+                buffers[rgba].positionLength += counts[i];
+                buffers[rgba].indexLength += indexCounts[i];
+                buffers[rgba].batchIds.push(i);
+            }
+        }
+
+        var object;
```

Comment:
Possibly `var buffer` is a better name, or something else? This is in a few places.